cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -1467,6 +1467,59 @@ impl<T: CompilePhase> MachBufferFinalized<T> {
     pub fn call_sites(&self) -> &[MachCallSite] {
         &self.call_sites[..]
     }
+
+    /// Get the disassembly of the buffer, using the given capstone context.
+    #[cfg(feature = "disas")]
+    pub fn disassemble(
+        &self,
+        params: Option<&crate::ir::function::FunctionParameters>,
+        cs: &capstone::Capstone,
+    ) -> Result<String, anyhow::Error> {
+        use std::fmt::Write;
+
+        let mut buf = String::new();
+
+        let relocs = self.relocs.as_slice();
+        let traps = self.traps.as_slice();
+
+        let insns = cs.disasm_all(&self.data, 0x0).map_err(map_caperr)?;
+        for i in insns.iter() {
+            write!(buf, "  ")?;
+
+            if let Some(s) = i.mnemonic() {
+                write!(buf, "{}\t", s)?;
+            }
+
+            if let Some(s) = i.op_str() {
+                write!(buf, "{}", s)?;
+            }
+
+            let end = i.address() + i.bytes().len() as u64;
+            let contains = |off| i.address() <= off && off < end;
+
+            if let Some(reloc) = relocs.iter().find(|reloc| contains(reloc.offset as u64)) {
+                write!(
+                    buf,
+                    " ; reloc_external {} {} {}",
+                    reloc.kind,
+                    reloc.name.display(params),
+                    reloc.addend,
+                )?;
+            }
+
+            if let Some(trap) = traps.iter().find(|trap| contains(trap.offset as u64)) {
+                write!(buf, " ; trap: {}", trap.code)?;
+            }
+
+            writeln!(buf, "")?;
```

Comment:
The format string is optional in the `writeln` macro; if you just want a newline you can leave it out.
```suggestion
            writeln!(buf)?;
```